llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2601,9 +2607,9 @@ BasicBlock *InnerLoopVectorizer::emitMemRuntimeChecks(BasicBlock *Bypass) {
 void InnerLoopVectorizer::createVectorLoopSkeleton(StringRef Prefix) {
   LoopVectorPreHeader = OrigLoop->getLoopPreheader();
   assert(LoopVectorPreHeader && "Invalid loop structure");
-  assert((OrigLoop->getUniqueExitBlock() ||
+  assert((OrigLoop->getUniqueLatchExitBlock() ||
           Cost->requiresScalarEpilogue(VF.isVector())) &&
-         "multiple exit loop without required epilogue?");
+         "loops not exiting via the latch without required epilogue?");
```

Comment:
Potentially committed separately?

Reply:
If possible I'd prefer not to make this patch dependent on another PR, since this PR has already been blocking https://github.com/llvm/llvm-project/pull/88385 for two months. Could some of these comments be addressed after this PR lands or in an independent PR?

Reply:
I'll split off the bits suggested by @ayalz and commit them separately without an extra PR given the changes have been approved and the benefit of splitting them off is to reduce the individual diffs (and make the changes clearly NFC)

Reply:
Perfect, thank you!